From 460fa290f162368134319a8dd5ca76b8597a0701 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Wed, 29 May 2024 00:03:54 +0800
Subject: [PATCH] Revert "backend-drm: Add mesa gbm to support imx93"

This reverts commit 2eb89930f839a516f26400cde7888fdb8cef860b.

This patch is not compatible with nxp-ls1028 platform, because
the libgbm is installed in usr/lib64.

Upstream-Status: Inappropriate [WRLinux specific]

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 libweston/backend-drm/drm-internal.h | 2 --
 libweston/backend-drm/drm.c          | 1 -
 libweston/backend-drm/kms.c          | 3 +--
 meson.build                          | 1 -
 4 files changed, 1 insertion(+), 6 deletions(-)

diff --git a/libweston/backend-drm/drm-internal.h b/libweston/backend-drm/drm-internal.h
index 7b274afb..4450666c 100644
--- a/libweston/backend-drm/drm-internal.h
+++ b/libweston/backend-drm/drm-internal.h
@@ -727,8 +727,6 @@ struct drm_output {
 	submit_frame_cb virtual_submit_frame;
 
 	enum wdrm_content_type content_type;
-
-	int (*surface_get_in_fence_fd)(struct gbm_surface *surface);
 };
 
 void
diff --git a/libweston/backend-drm/drm.c b/libweston/backend-drm/drm.c
index ea4d96bd..9ab72e44 100644
--- a/libweston/backend-drm/drm.c
+++ b/libweston/backend-drm/drm.c
@@ -2669,7 +2669,6 @@ drm_output_create(struct weston_backend *backend, const char *name)
 	output->max_bpc = 16;
 #ifdef BUILD_DRM_GBM
 	output->gbm_bo_flags = GBM_BO_USE_SCANOUT | GBM_BO_USE_RENDERING;
-	output->surface_get_in_fence_fd = weston_load_module("libgbm.so", "gbm_surface_get_in_fence_fd", LIBDIR);
 #endif
 
 	weston_output_init(&output->base, b->compositor, name);
diff --git a/libweston/backend-drm/kms.c b/libweston/backend-drm/kms.c
index 6621c678..b6b8eca9 100644
--- a/libweston/backend-drm/kms.c
+++ b/libweston/backend-drm/kms.c
@@ -1189,8 +1189,7 @@ drm_output_apply_state_atomic(struct drm_output_state *state,
 		 * the buffer_release was not exist or
 		 * the buffer was not used in the output.
 		 */
-		if (output->surface_get_in_fence_fd)
-			in_fence_fd = output->surface_get_in_fence_fd(output->gbm_surface);
+		in_fence_fd = gbm_surface_get_in_fence_fd(output->gbm_surface);
 	}
 #if defined(ENABLE_IMXG2D)
 	else if(b->use_g2d && b->g2d_renderer) {
diff --git a/meson.build b/meson.build
index 53dd670b..ad2240c5 100644
--- a/meson.build
+++ b/meson.build
@@ -113,7 +113,6 @@ config_h.set_quoted('PACKAGE_BUGREPORT', 'https://gitlab.freedesktop.org/wayland
 
 config_h.set_quoted('BINDIR', dir_bin)
 config_h.set_quoted('DATADIR', dir_data)
-config_h.set_quoted('LIBDIR', dir_lib)
 config_h.set_quoted('LIBEXECDIR', dir_libexec)
 config_h.set_quoted('MODULEDIR', dir_module_weston)
 config_h.set_quoted('LIBWESTON_MODULEDIR', dir_module_libweston)
-- 
2.34.1

